cmd/kpromo/cmd/pr/pr.go

```diff
@@ -0,0 +1,441 @@
+		}
+
+		// If the manifest was not modified, exit now
+		if string(newlist) == string(oldlist) {
+			logrus.Info("No changes detected in the promoter images list, exiting without changes")
+			return nil
+		}
+	}
+
+	// add the modified manifest to staging
+	logrus.Debugf("Adding %s to staging area", imagesListPath)
+	if err := repo.Add(imagesListPath); err != nil {
+		return errors.Wrap(err, "adding image manifest to staging area")
+	}
+
+	commitMessage := "releng: Image promotion for " + opts.project + " " + strings.Join(opts.tags, " / ")
+
+	// Commit files
+	logrus.Debug("Creating commit")
+	if err := repo.UserCommit(commitMessage); err != nil {
+		return errors.Wrapf(err, "Error creating commit in %s/%s", git.DefaultGithubOrg, k8sioRepo)
+	}
+
+	// Push to fork
+	if mustRun(opts, fmt.Sprintf("Push changes to user's fork at %s/%s?", userForkOrg, userForkRepo)) {
+		logrus.Infof("Pushing manifest changes to %s/%s", userForkOrg, userForkRepo)
+		if err := repo.PushToRemote(userForkName, branchname); err != nil {
+			return errors.Wrapf(err, "pushing %s to %s/%s", userForkName, userForkOrg, userForkRepo)
+		}
+	} else {
+		// Exit if no push was made
+
+		logrus.Infof("Exiting without creating a PR since changes were not pushed to %s/%s", userForkOrg, userForkRepo)
+		return nil
+	}
+
+	// Create the Pull Request
+	if mustRun(opts, "Create pull request?") {
+		pr, err := gh.CreatePullRequest(
+			git.DefaultGithubOrg, k8sioRepo, k8sioDefaultBranch,
+			fmt.Sprintf("%s:%s", userForkOrg, branchname),
+			commitMessage, generatePRBody(opts),
+		)
+		if err != nil {
+			return errors.Wrap(err, "creating the pull request in k/k8s.io")
+		}
+		logrus.Infof(
+			"Successfully created PR: %s%s/%s/pull/%d",
+			github.GitHubURL, git.DefaultGithubOrg, k8sioRepo, pr.GetNumber(),
+		)
+	}
+
+	// Success!
+	return nil
+}
+
+// mustRun avoids running when a users chooses n in interactive mode
+func mustRun(opts *promoteOptions, question string) bool {
+	if !opts.interactiveMode {
+		return true
+	}
+	_, success, err := util.Ask(fmt.Sprintf("%s (Y/n)", question), "y:Y:yes|n:N:no|y", 10)
+	if err != nil {
+		logrus.Error(err)
+		if err.(util.UserInputError).IsCtrlC() {
+			os.Exit(1)
+		}
+		return false
+	}
+	if success {
+		return true
+	}
+	return false
+}
+
+// generatePRBody creates the body of the Image Promotion Pull Request
+func generatePRBody(opts *promoteOptions) string {
+	args := fmt.Sprintf("--fork %s", opts.userFork)
+	if opts.interactiveMode {
+		args += " --interactive"
+	}
+
+	if opts.project != defaultProject {
+		args += " --project" + opts.project
+	}
+
+	if opts.reviewers != defaultReviewers {
+		args += " --reviewers" + opts.reviewers
+	}
+
+	for _, tag := range opts.tags {
+		args += " --tag " + tag
+	}
+
+	prBody := fmt.Sprintf("Image promotion for %s %s\n", opts.project, strings.Join(opts.tags, " / "))
+	prBody += "This is an automated PR generated from `kpromo`\n"
+	prBody += fmt.Sprintf("```\nkpromo pr %s\n```\n\n", args)
```

Comment:
Nice detail capturing the command here

Reply:
> Nice detail capturing the command here

Heh, I can't take credit!
That was @palnabarun in https://github.com/kubernetes/release/pull/2320 and @CecileRobertMichon in https://github.com/kubernetes/release/pull/2280. :)